ngraph/frontend/onnx_import/include/onnx_import/editor/detail/subgraph_extraction.hpp

```diff
@@ -0,0 +1,164 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <map>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace ONNX_NAMESPACE
+{
+    class GraphProto;
+    class NodeProto;
+    class ValueInfoProto;
+} // namespace ONNX_NAMESPACE
+
+namespace ngraph
+{
+    enum class EdgeType
+    {
+        INPUT,
+        OUTPUT
+    };
+
+    template <EdgeType>
+    struct Edge
+    {
+        Edge() = delete;
+        Edge(const int node_idx, std::string tensor_name)
+            : m_node_idx{node_idx}
+            , m_tensor_name{std::move(tensor_name)}
+        {
+        }
+
+        const int m_node_idx;
```

Comment:
How can one obtain a valid value for m_node_idx? Briefly looking in the code I see that a valid value is really required but there is no way for use to deduce this index based on some user-visible name without parsing model proto file. Even Netron doesn't give this index.

Can we make it at least optional? In this case it would mean cutting by a tensor name and providing a single input for all the consumers.

Reply:
Similarily to the comment below - TBD in 47578